stdlib/src/main/scala/chisel3/std/tilelink/Data.scala

```diff
@@ -0,0 +1,101 @@
+package chisel3.std.tilelink
+
+import chisel3._
+import chisel3.util.DecoupledIO
+
+import scala.collection.immutable.SeqMap
+
+private class NoTLCException(
+  channel:       String,
+  linkParameter: TileLinkLinkParameter)
+    extends ChiselException(
+      s"call $channel in TLBundle is not present in a TL-UL or TL-UH bus:\n $linkParameter"
+    )
+
+class TLBundle(val parameter: TileLinkLinkParameter) extends Record with experimental.AutoCloneType {
+  def a: DecoupledIO[TLChannelA] =
+    elements("a").asInstanceOf[DecoupledIO[TLChannelA]]
+
+  def b: DecoupledIO[TLChannelB] =
+    elements
+      .getOrElse("b", throw new NoTLCException("b", parameter))
+      .asInstanceOf[DecoupledIO[TLChannelB]]
+
+  def c: DecoupledIO[TLChannelC] =
+    elements
+      .getOrElse("c", throw new NoTLCException("c", parameter))
+      .asInstanceOf[DecoupledIO[TLChannelC]]
+
+  def d: DecoupledIO[TLChannelD] =
+    elements("d").asInstanceOf[DecoupledIO[TLChannelD]]
+
+  def e: DecoupledIO[TLChannelE] =
+    elements
+      .getOrElse("e", throw new NoTLCException("e", parameter))
+      .asInstanceOf[DecoupledIO[TLChannelE]]
+
+  val elements: SeqMap[String, DecoupledIO[Bundle]] =
+    SeqMap[String, DecoupledIO[Bundle]](
+      "a" -> DecoupledIO(new TLChannelA(parameter.channelAParameter)),
+      "d" -> Flipped(DecoupledIO(new TLChannelD(parameter.channelDParameter)))
+    ) ++ (
+      if (parameter.hasBCEChannels)
+        Seq(
+          "b" -> Flipped(DecoupledIO(new TLChannelB(parameter.channelBParameter.get))),
+          "c" -> DecoupledIO(new TLChannelC(parameter.channelCParameter.get)),
+          "e" -> DecoupledIO(new TLChannelE(parameter.channelEParameter.get))
+        )
+      else
+        Seq()
+    )
+}
+
+class TLChannelA(val parameter: TileLinkChannelAParameter) extends Bundle {
+  private val maskWidth = parameter.dataWidth / 8
+  // TODO: in spec, this is 'code' in spec
```

Comment:
What does this comment mean?

Reply:
There is a probable typo in TileLink spec, where the operation code field in channel A is called `a_code` but those in channel B,C,D,E are called `b_opcode` etc. (c.f. Tilelink spec v 1.8.1 page 15). I have updated the comment to clarify the intention hopefully.